src/main/java/calculator/extractor/CustomDelimiterTextExtractor.kt

```diff
@@ -0,0 +1,34 @@
+package calculator.extractor
+
+import calculator.Number
+import calculator.Numbers
+import calculator.exception.TextDelimiterExtractorException
+import java.util.regex.Pattern
+
+class CustomDelimiterTextExtractor : TextDelimiterExtractor {
+
+    override fun isSupport(text: String?): Boolean {
+        return text?.let { PATTERN.matcher(it).find() } ?: false
+    }
+
+    override fun extract(text: String): Numbers {
+        val matcher = PATTERN.matcher(text)
+
+        return if (matcher.find()) {
+            val customDelimiter = matcher.group(1)
+
+            Numbers(
+                matcher.group(2)
+                    .split(customDelimiter)
+                    .map { Number(it) }
+            )
+        } else {
+            throw TextDelimiterExtractorException("Unsupported text format: $text")
+        }
+    }
+
+    companion object {
+        private const val REGEX_PATTERN: String = "//(.)\n(.*)"
+        private val PATTERN = Pattern.compile(REGEX_PATTERN)
```

Comment:
해당 상수가 어떤 정규식인지 네이밍으로 드러나면 좋을 것 같아요. 

추가적으로 Pattern 객체 프로퍼티로 들어갈 정규식은 굳이 상수로 뽑지 않아도 Pattern 객체 상수를 통해서 충분히 의도를 이해할 수 있을 것 같아요!

```suggestion
        private val PATTERN = Pattern.compile("//(.)\n(.*)")
```